### PR TITLE
Introduce Clippy Github Actions

### PR DESCRIPTION
**Pull request by kinggoesgaming**
**I'm submitting a(n)** other

# Description
Moves clippy tests over to Github Actions

### CODE REVIEW

**Comment by kinggoesgaming** [comment]
You will notice some `#[allow(..)]`s. we need to address them at some point and figure out if they need to exist or not

**Comment by KodrAus** [comment]
Do you think we need to run clippy on multiple platforms?

**Comment by kinggoesgaming** [comment]
I think in particular we need one for Windows as guid feature is a thing

I can remove macOS build, but quite honestly currently the build times are not too bad

**Comment by kinggoesgaming** [comment]
@KodrAus / @DylanDPC this is ready to go

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+       with:
+         components: clippy
+         override: true
+         profile: minimal
+         toolchain: 1.32.0
+     - name: Run `cargo clippy`
+       uses: actions-rs/cargo@v1
+       with:
+         command: clippy
+         args: --features "guid serde slog std v1 v3 v4 v5" -- -D warnings
```
Comment (Dylan-DPC) [inline .github/workflows/clippy.yml]: do we need all the features activated for windows builds as well?

Comment (kinggoesgaming) [reply .github/workflows/clippy.yml]: Tbh it doesn’t affect the overall build time much... we can see later if we need to switch it around

**Comment by Dylan-DPC** [comment]
bors: r+

**Pull request merged**